### PR TITLE
Suppress unnecessary timeline events for empty record creation and deletion

### PR DESCRIPTION
**Pull request by yadavshubham01**
This pull request introduces changes to the TimelineActivityRepository class, specifically in the upsertOne method, to enhance the timeline management by suppressing unnecessary events related to the creation and deletion of empty records. The goal is to maintain a cleaner timeline feed by ensuring that only meaningful changes are logged.

Changes Made
Event Suppression Logic:

Added a new method shouldSuppressEvent to determine whether a timeline event should be suppressed based on recent activities. This method checks:
If there is recent activity related to the same record.
If both the recent and current events are either creation or deletion events.
If both events pertain to the same object type.
Handling of Empty Records:

In the upsertOne method, if the event being processed is a creation or deletion and it is determined that the associated record is empty (i.e., no fields have been edited), the event will be suppressed.
If a suppression is triggered, any recent timeline activity related to that record will be removed to prevent clutter in the timeline.
Maintaining Existing Functionality:

The existing logic for updating and inserting timeline activities remains intact, ensuring that meaningful changes are still logged appropriately.
Benefits
Cleaner Timeline Feed: By suppressing logs for empty record creation and deletion, the timeline becomes more meaningful, focusing only on significant changes.
Improved User Experience: Users will no longer see repetitive and unnecessary notifications about empty records, leading to a better overall experience when interacting with the timeline.

Conclusion
These changes aim to enhance the functionality of the timeline feature by ensuring that it accurately reflects user actions without unnecessary noise. Your review and feedback on this implementation would be greatly appreciated.

### CODE REVIEW

**Comment by yadavshubham01** [comment]
hey @FelixMalfait @Bonapara please review this PR

**Comment by yadavshubham01** [comment]
hey @FelixMalfait @charlesBochet please review this pr

File: install.sh
Comment (FelixMalfait) [inline install.sh:1]: Hey, qhy change the chmod here? Thanks

File: packages/twenty-server/src/modules/timeline/repositiories/timeline-activity.repository.ts [hunk 1]
```diff
@@ -37,6 +37,35 @@ export class TimelineActivityRepository {
       workspaceId,
     );
 
+    private getObjectType(eventName: string): string {
```
Comment (FelixMalfait) [inline packages/twenty-server/src/modules/timeline/repositiories/timeline-activity.repository.ts:40]: You put a function within a function?

**Comment by FelixMalfait** [comment]
Hey @yadavshubham01 sorry I'll close this because it doesn't work and I think it needs a bigger refactor. Thanks

**Pull request closed without merging**